### DISCUSSION TITLE
How can I scan a running pod or /proc directory with Trivy?

### DISCUSSION THREAD

**Discussion by pohanhuangtw** [aquasecurity/trivy#8187] [Q&A]
### Goal
I’d like to scan my running pod in Kubernetes for vulnerabilities, or scan the /proc directory specifically.

### What You’ve Tried: 

- trivy fs /proc 
- trivy rootfs /proc

Both of them get nothing (I am scanning the nginx pod, I am not try to run trivy image, but just run the scan running pod)

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

None

**Comment by itaysk** [comment]
Trivy can scan the Pod's image (even automatically with `trivy k8s` command) but if I understand correctly that's not what you're looking for ("I am not try to run trivy image, but just run the scan running pod").
If you want to scan the live filesystem of the pod, you would need to find a way to run trivy in that pod (or mount it's filesystem).
If you want to scan the entire host you can run trivy on the [filesystem root](https://trivy.dev/latest/docs/target/rootfs/).
You can't scan `/proc` since Trivy [doesn't](https://github.com/aquasecurity/trivy/discussions/4184#discussioncomment-5827037) [follow symlinks](https://github.com/aquasecurity/trivy/issues/7456) at the moment.

**Comment by pohanhuangtw** [comment]
Hi @itaysk ,
Thanks for the response, is there any recommendation to workaround it? If I want to generate sbom from a /proc?

**Reply by itaysk** [reply]
no, but if you explain what you are trying to achieve maybe there's another way

**Reply by pohanhuangtw** [reply]
Sure, I am trying to generate or scan a workload based on its PID. I attempted to use trivy fs for this purpose. I discovered that I can mount /proc/<pid>/root/ and then run trivy fs on it.

However, I’m not sure if there’s a better approach or if I should modify the code to achieve this. Do you have any suggestions?

**Reply by itaysk** [reply]
sorry, we can't help you with that. you'll need to find a way to map the pid to something trivy can scan (image, filesystem, etc)

**Reply by pohanhuangtw** [reply]
Thanks @itaysk .
I have another question about trivy, Is Trivy's database artifact also licensed under Apache 2.0?

**Reply by itaysk** [reply]
unfortunately the answer to this question remains unclear. as far as we (Aqua/Trivy) concern the DB is part of Trivy OSS product, but last time we looked into it some specific data sources used by trivy-db had unclear licenses, so although we could redistribute them, we couldn't relicense them into Apache2.
